controllers/provider-aws/pkg/cmd/config.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/config"
+	configinstall "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/config/install"
+	configv1alpha1 "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/config/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"github.com/spf13/pflag"
+)
+
+// ConfigOptions are command line options that can be set for config.ControllerConfiguration.
+type ConfigOptions struct {
+	// Kubeconfig is the path to a kubeconfig.
+	ConfigFilePath string
+
+	scheme *runtime.Scheme
+	codecs serializer.CodecFactory
+	config *Config
+}
+
+// Config is a completed controller configuration.
+type Config struct {
+	// Config is the controller configuration.
+	Config *config.ControllerConfiguration
+}
+
+func (c *ConfigOptions) buildConfig(scheme *runtime.Scheme, codecs serializer.CodecFactory) (*config.ControllerConfiguration, error) {
+	if len(c.ConfigFilePath) == 0 {
+		return nil, fmt.Errorf("config file path not set")
+	}
+
+	c.scheme = scheme
```

Comment:
In principle, it's a kind of an anti-pattern to initialize fields in a method different from a constructor (or one invoked by a constructor). In this particular case, `scheme` and `codecs` should either not be fields (so passed explicitly to every method that needs them), or if they remain fields, they should be initialized before calling `Complete`.